accepted/future-releases/nnbd/feature-specification.md

```diff
@@ -887,6 +890,42 @@ was marked `late`.  Note that this is a change from pre-NNBD semantics in that:
   - Reading the variable during initializer evaluation is no longer checked for,
     and does not cause an error.
 
+### Boolean conditional evaluation.
+
+The requirement that the condition in a boolean conditional control expression
+(e.g. the a conditional statement, conditional element, `while` loop, etc) be
+assignable to `bool` is unchanged from pre null-safe Dart.  The change in
+assignability means that the static type of the condition may only be `dynamic`,
+`Never`, or `bool`.  In full null-safe Dart, an expression of type `Never` will
+always diverge and an expression of type `bool` will never evaluate to a value
+other than `true` or `false`, and hence no conversion is required in these
+cases.  A conditional expression of type `dynamic` may evaluated to any value,
+and hence must be implicitly downcast to `bool`, after which no further check is
+required.
+
+In mixed mode (weak mode evaluation) Dart, values of type `Never` and `bool` may
```

Comment:
> In mixed mode (weak mode evaluation) Dart

It doesn't matter if it is mixed or not. Maybe simply 'In weak checking mode...'

Reply:
Done.